core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1334,6 +1334,27 @@ class UnifiedLog(@volatile var logStartOffset: Long,
         } else {
           Some(new TimestampAndOffset(RecordBatch.NO_TIMESTAMP, -1L, Optional.of(-1)))
         }
+      } else if (targetTimestamp == ListOffsetsRequest.EARLIEST_PENDING_UPLOAD_OFFSET_TIMESTAMP) {
```

Comment:
Can we extract this into different method and call it from here especially that the logic here has too many if stataments ? I know the rest of fetchOffsetByTimestamp has been done in-place but the method has grown and it hard to follow.

Reply:
Done